web/src/HUD.tsx

```diff
@@ -139,15 +157,25 @@ class HUD extends Component<HudProps, HudState> {
       combinedLog = view.Log
     }
 
+    let previewRoute = (props: RouteComponentProps<any>) => {
+      let name = props.match.params ? props.match.params.name : ""
+      let endpoint = ""
+      if (view && name !== "") {
+        let r = view.Resources.find(r => r.Name === name)
+        endpoint = r ? r.Endpoints && r.Endpoints[0] : ""
```

Comment:
if there's no endpoint, we probably don't want to return a PreviewPane at all? maybe better to fallback to a LogPane?